types/optional.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import "encoding/json"
+
+// Optional struct leverages any type as optional and nullable.
+// Optional.set is true when Optional.value is to be used.
+type Optional[T any] struct {
+	value *T
+	set   bool // set is true when its value is to be used
+}
+
+func NewOptional[T any](value *T) Optional[T] {
+	return Optional[T]{
+		value: value,
+		set:   true,
+	}
+}
+
+func EmptyOptional() Optional[any] {
+	return Optional[any]{
+		value: nil,
+		set:   false,
+	}
+}
+
+func (o *Optional[T]) Value() T {
+	return *o.value
+}
+
+func (o *Optional[T]) SetValue(value T) {
+	o.value = &value
```

Comment:
SetValue() should get a pointer of T type. Nil value cannot be assigned without pointer variable.

Reply:
Okay, thanks for pointing out